client/quantum_serverless/core/program.py

```diff
@@ -81,11 +79,13 @@ def from_json(cls, data: Dict[str, Any]):
         return Program(**{k: v for k, v in data.items() if k in field_names})
 
     def __post_init__(self):
-        if self.arguments is not None and not is_jsonable(
-            self.arguments, cls=QiskitObjectsEncoder
-        ):
-            raise QuantumServerlessException(
-                "Arguments provided are not json serializable."
+        if self.arguments is not None:
+            warnings.warn(
+                "Passing `arguments` as constructor argument to `Program` is deprecated. "
```

Comment:
nit: do we have an idea of when we're going to remove this? if we do, it might make sense to include that here (e.g. "it will be removed in v0.0.9" or something like that). but not a big deal if not either

Reply:
Agree!